packages/use-intl/types/index.d.ts

```diff
@@ -1,7 +1,15 @@
 // This type is intended to be overridden
-// by the consumer for optional type safety
+// by the consumer for optional type safety of messages
 declare interface IntlMessages extends Record<string, any> {}
 
+// This type is intended to be overridden
+// by the consumer for optional type safety of formats
+declare interface IntlFormats {
+  dateTime: {};
+  number: {};
+  list: {};
```

Comment:
I think we could avoid all the `// @ts-expect-error` declarations in the test code by using a more flexible definition like this:

```suggestion
  dateTime: any;
  number: any;
  list: any;
```

- [ ] Avoid `@ts-expect-error`